### PR TITLE
[JIT] [Remove Mutation] Add handling of normal_

### LINKED ISSUE #51735: aten::normal_ not handled as a special op in RemoveTensorMutation pass.
## 🐛 Bug

aten::normal_ should be considered as a special op in remove mutation pass.
The op does not have an equivalent aten::normal

## To Reproduce

Apply remove mutation on the model below:

```
class Noise(nn.Module):
    def forward(self, image):
        return image.new_empty(2, 1, 3, 4).normal_()
```

## Expected behavior

Error is:
RuntimeError: 0 INTERNAL ASSERT FAILED at "/pytorch/torch/csrc/jit/ir/alias_analysis.cpp":532, please report a bug to PyTorch. We don't have an op for aten::normal but it isn't a special case.  Argument types: Tensor, float, float, None, 

cc @ezyang @bhosmer @smessmer @ljk53 @bdhirsh @ailzhang @gmagogsfm

### PR DESCRIPTION
**Pull request by eellison**
Fixes https://github.com/pytorch/pytorch/issues/51735

### CODE REVIEW

File: torch/csrc/jit/passes/remove_mutation.cpp [hunk 1]
```diff
@@ -35,6 +35,29 @@ Node* MutationRemover::createSpecialMappedOp(Node* n) {
...
   } else if (n->matches("aten::zero_(Tensor(a!) self) -> Tensor(a!)")) {
     new_node = graph_->insert(aten::zeros_like, {n->inputs().at(0)})->node();
+  } else if (
+      n->matches(
+          "aten::normal_(Tensor(a!) self, float mean=0, float std=1, *, Generator? generator=None) -> Tensor(a!)")) {
+    // TODO: we should have normal_like operator
+    // normal(float mean, float std, int[] size, *, Generator? generator=None,
+    // ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool?
+    // pin_memory=None) -> Tensor
+    auto size = graph_->insert(aten::size, {n->inputs().at(0)});
```
Comment (Krovatkin) [inline torch/csrc/jit/passes/remove_mutation.cpp:45]: Is this transformation always on in `RemoveMutation` pass? I feel a little bit uneasy seeing how many extra ops we are adding here?

Comment (eellison) [reply torch/csrc/jit/passes/remove_mutation.cpp:45]: Yes, but `RemoveMutation` isn't always on anywhere

Comment (eellison) [reply torch/csrc/jit/passes/remove_mutation.cpp:45]: I opened issue here: https://github.com/pytorch/pytorch/issues/52334. I dont think it should block landing this bug fix.

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Pull request closed without merging**